cli/azd/pkg/tools/azcli/container_registry.go

```diff
@@ -85,6 +87,27 @@ func (crs *containerRegistryService) GetContainerRegistries(
 }
 
 func (crs *containerRegistryService) Login(ctx context.Context, subscriptionId string, loginServer string) error {
+	dockerCreds, err := crs.Credentials(ctx, subscriptionId, loginServer)
+	if err != nil {
+		return err
+	}
+
+	err = crs.docker.Login(ctx, dockerCreds.LoginServer, dockerCreds.Username, dockerCreds.Password)
+	if err != nil {
+		return fmt.Errorf(
+			"failed logging into docker registry %s: %w",
+			loginServer,
+			err)
+	}
+
+	return nil
+}
+
+// Credentials gets the credentials that could be used to login to the specified container registry. It prefers to use
+// AAD token credentials for the current user, but if that fails it will fall back to admin user credentials.
+func (crs *containerRegistryService) Credentials(
```

Comment:
It feels a little weird that one would call `Credentials(loginServer)` which then returns a `DockerCredentials{loginServer}`. Should the reader assume that the returned `loginServer` is different than the one passed in? I understand why this shows up this way because of the "container helper" indirection, but just wanted to note down an observation on the design.

Reply:
I will look into trying to clean this up - either by removing the login server from the return value (assuming that it is not too destructive to the rest of the code base) or by adding a comment to the method noting that the loginServer will always match the parameter on success and is added as a convience.